l1-verifier/src/Trollup.sol

```diff
@@ -40,4 +90,25 @@ contract Trollup is Verifier {
             revert InvalidSNARK();
         }
     }
+
+    function processTx(TxProof memory l2Tx) internal {
+        uint256 txKind = l2Tx.input[11];
+        if (txKind == 0) {
+            // L2 transfer
+            /* nothing to do on L1 */
+        } else if (txKind == 1) {
+            // L1 deposit
+            uint256 l2TxValue = l2Tx.input[7];
+            uint256 l2Recipient = l2Tx.input[15];
+            if (deposits[l2Recipient].available == 0 || l2TxValue == 0) revert InvalidDeposit();
```

Comment:
I think we don't need this line. The `available` case is already handled by the next line, and if the sequencer wants to waste gas on deposits of 0 value it's their problem.